src/PointwiseFunctions/GeneralRelativity/ComputeGhQuantities.hpp

```diff
@@ -196,4 +200,154 @@ Scalar<DataType> time_deriv_of_lapse(
     const tnsr::A<DataType, SpatialDim, Frame>& spacetime_unit_normal,
     const tnsr::iaa<DataType, SpatialDim, Frame>& phi,
     const tnsr::aa<DataType, SpatialDim, Frame>& pi) noexcept;
+
+/*!
+ * \ingroup GeneralRelativityGroup
+ * \brief Computes spatial derivatives of the shift vector from
+ *        the generalized harmonic and geometric variables
+ *
+ * \details Let the generalized harmonic spatial derivative variable be
+ * \f$\Phi_{iab} = \partial_i \psi_{ab} \f$. The spatial derivatives of the
+ * shift vector \f$N^i\f$ are:
+ * \f{align*}
+ *  \psi^{ja} N t^b \Phi_{iab} - 2 t^j \partial_i N
+ *  =& (g^{ja} - t^j t^a) N t^b \Phi_{iab} - 2 t^j \partial_i N \\
+ *  =& N g^{ja} t^b \partial_i \psi_{ab}
+ * \f}
+ * where we used the equation from above for \f$\partial_i N\f$. Then
+ * \f{align*}
+ * \psi^{ja} N t^b \Phi_{iab} - 2 t^j \partial_i N
+ *  =& g^{ja} (\partial_i \psi_{a0} - N^k \partial _i \psi_{ak}) \\
+ *  =& g^{jl} (\partial_i N_l - N^k \partial_i g_{lk}) (\because g^{j0} = 0) \\
+ *  =& g^{jl} (N^k \partial_i g_{lk}
+ *             + g_{kl}\partial_i N^k - N^k \partial_i g_{kl}) \\
+ *  =& g^{jl} g_{kl} \partial_i N^k \\
+ *  =& \partial_i N^j
+ * \f}
+ * I.E.
+ * \f{align*}
+ *  \partial_i N^j =& N\psi^{ja} t^b \Phi_{iab} - 2 t^j \partial_i N \\
+ *                 =& N (\psi^{ja} + t^j t^a) t^b \Phi_{iab}.
+ * \f}
+ */
+template <size_t SpatialDim, typename Frame, typename DataType>
+void spatial_deriv_of_shift(
+    gsl::not_null<tnsr::iJ<DataType, SpatialDim, Frame>*> deriv_shift,
+    const Scalar<DataType>& lapse,
+    const tnsr::AA<DataType, SpatialDim, Frame>& inverse_spacetime_metric,
+    const tnsr::A<DataType, SpatialDim, Frame>& spacetime_unit_normal,
+    const tnsr::iaa<DataType, SpatialDim, Frame>& phi) noexcept;
+
+template <size_t SpatialDim, typename Frame, typename DataType>
+tnsr::iJ<DataType, SpatialDim, Frame> spatial_deriv_of_shift(
+    const Scalar<DataType>& lapse,
+    const tnsr::AA<DataType, SpatialDim, Frame>& inverse_spacetime_metric,
+    const tnsr::A<DataType, SpatialDim, Frame>& spacetime_unit_normal,
+    const tnsr::iaa<DataType, SpatialDim, Frame>& phi) noexcept;
+
+/*!
+ * \ingroup GeneralRelativityGroup
+ * \brief Computes time derivative of the shift vector from
+ *        the generalized harmonic and geometric variables
+ *
+ * \details Let the generalized harmonic spatial derivative variable be
+ * \f$\Phi_{iab} = \partial_i \psi_{ab} \f$. The time derivative of \f$ N^i \f$
+ * is given by:
+ * \f{align*}
+ *  -N^2 t^b \Pi_{kb} g^{ik} + N N^j t^b \Phi_{jkb} g^{ik}
```

Comment:
This equation would also be more readable if it were formatted with similar recommendations to the above (effectively in reverse).

Reply:
reversed